pennylane/_qubit_device.py

```diff
@@ -281,16 +281,17 @@ def execute(self, circuit, **kwargs):
         if self.shots is not None or any(isinstance(m, sample_type) for m in circuit.measurements):
             # Lightning does not support apply(rotations) anymore, so we need to rotate here
             is_lightning = (
-                hasattr(self, "name") and isinstance(self.name, str) and "Lightning" in self.name
+                hasattr(self, "name")
+                and isinstance(self.name, str)
+                and "Lightning" in self.name
+                and getattr(self, "_CPP_BINARY_AVAILABLE", False)
             )
             diagonalizing_gates = self._get_diagonalizing_gates(circuit) if is_lightning else None
             if is_lightning and diagonalizing_gates:
-                self.apply_lightning(diagonalizing_gates)
+                self.apply(diagonalizing_gates)
```

Comment:
Just curious, why the change from `apply_lightning()` to `apply()`?

Reply:
Ah, now I see it is related to failure [here](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/8209817284/job/22456073311#step:12:333).